### DISCUSSION TITLE
System.NullReferenceException

### DISCUSSION THREAD

**Discussion by LumiProj** [vitalyster/SharpXMPP#148] [General]
```
{
  "length": 0,
  "offset": 0,
  "id": "fd75c885-3fc1-4b52-8943-a7e28d1f2b30",
  "exception": {
    "type": "System.NullReferenceException",
    "message": "Object reference not set to an instance of an object",
    "stackTrace": "  at SharpXMPP.XmppTcpConnection.InitTlsIfSupported (SharpXMPP.XMPP.Stream.Elements.Features features) <0x1065ec560 + 0x00038> in <af879d82b0fe4cf8b1d541279e6c3414#0835d9c74997d27a0e543614112fcf97>:0 \n  at SharpXMPP.XmppTcpConnection.ConnectAsync (System.Threading.CancellationToken token) <0x1065ea9a0 + 0x0046f> in <af879d82b0fe4cf8b1d541279e6c3414#0835d9c74997d27a0e543614112fcf97>:0 \n  at LumiDocTalk.ViewModel.VoipView_Model.ConnectXmppServer (System.String username, System.String password) <0x102ba6490 + 0x008bf> in <8a5d829cabcf4ba38d7fff223dc2c7f5#0835d9c74997d27a0e543614112fcf97>:0 ",
    "wrapperSdkName": "appcenter.xamarin"
  },
  "appId": "0dbe2aac-d460-40c0-b8f7-d9d3254b8f7c",
  "installId": "0B2BAEF7-5924-41AD-8789-1924E71E5E2F",
  "isTestMessage": false,
  "timestamp": "2021-07-02T20:59:30.639Z",
  "sid": "0a528ea5-884e-47d6-a4eb-a6643720fe69",
  "device": {
    "sdkName": "appcenter.ios",
    "sdkVersion": "4.1.1",
    "wrapperSdkVersion": "4.2.0",
    "wrapperSdkName": "appcenter.xamarin",
    "model": "iPhone12,3",
    "oemName": "Apple",
    "osName": "iOS",
    "osVersion": "14.6",
    "osBuild": "18F72",
    "locale": "en_PK",
    "timeZoneOffset": 300,
    "screenSize": "2436x1125",
    "appVersion": "10.0",
    "appBuild": "0.0.0.5",
    "appNamespace": "----------",
    "wrapperRuntimeVersion": "12.6.0"
  }
}
```

**Comment by ForNeVeR** [comment]
I think it is possible if `GetServerFeatures` returns `null`. Could you please debug it? And/or check the traffic received by the client?

**Comment by LumiProj** [comment]
Hi,
Please guide me is this issue is resolved.
This library is working fine in other iPhone models except this one. I have attached the device detail. I don't think this is any issue with Features it looks like a bug please guide me how to resolve this issue.
Almost 3 months have passed and didn't get the help.



Thanks

**Comment by ForNeVeR** [comment]
Please help us by debugging the issue. 3 months have passed, and you haven't answered any questions you've been asked, and haven't posted any debugging details about the issue.

Unfortunately, without the additional information I requested, we have no way of helping you.

**Comment by LumiProj** [comment]
Hi,

Unfortunately this issue is facing by my client and its hard to fetch the device debug mode details. in my development I have never face this problem everything is working fine but my few clients are facing this issue and I can only catch issue report using App Center I have already posted the AppCenter report. I know its hard to resolve the issue until and unless we know the debug mode details but in my development environment everything is in good shape.
I can only say my client is using iOS 7 device with latest sdk.

Following is the screenshot of App Center.
![image](https://user-images.githubusercontent.com/59764810/133305526-3b4a369b-b728-47d1-b3ef-4988b4173a16.png)


Following is the App Center report.

{
  "length": 0,
  "offset": 0,
  "id": "5afffac2-d9e6-46e3-bfe7-33d0ff3b1531",
  "exception": {
    "type": "System.NullReferenceException",
    "message": "Object reference not set to an instance of an object",
    "stackTrace": "  at SharpXMPP.XmppTcpConnection.InitTlsIfSupported (SharpXMPP.XMPP.Stream.Elements.Features features) <0x104360e40 + 0x00038> in <af879d82b0fe4cf8b1d541279e6c3414#77cfb59c0c3acdcd743eb6822b84caa5>:0 \n  at SharpXMPP.XmppTcpConnection.ConnectAsync (System.Threading.CancellationToken token) <0x10435f280 + 0x0046f> in <af879d82b0fe4cf8b1d541279e6c3414#77cfb59c0c3acdcd743eb6822b84caa5>:0 \n  at LumiDocTalk.ViewModel.VoipView_Model.ConnectXmppServer (System.String username, System.String password) <0x1009534a0 + 0x0098f> in <ce78fee37b764665bab154f1f93b29ab#77cfb59c0c3acdcd743eb6822b84caa5>:0 ",
    "wrapperSdkName": "appcenter.xamarin"
  },
  "appId": "0dbe2aac-d460-40c0-b8f7-d9d3254b8f7c",
  "installId": "A719F44F-64F6-47CF-91DD-350FB1E3F0B4",
  "isTestMessage": false,
  "timestamp": "2021-09-06T21:26:16.391Z",
  "sid": "8adae3ae-964e-4730-a0d8-7cc352b4bd07",
  "device": {
    "sdkName": "appcenter.ios",
    "sdkVersion": "4.1.1",
    "wrapperSdkVersion": "4.2.0",
    "wrapperSdkName": "appcenter.xamarin",
    "model": "iPhone9,2",
    "oemName": "Apple",
    "osName": "iOS",
    "osVersion": "14.7.1",
    "osBuild": "18G82",
    "locale": "en_PK",
    "timeZoneOffset": 300,
    "screenSize": "2208x1242",
    "appVersion": "11.0",
    "carrierName": "T.....",
    "carrierCountry": "pk",
    "appBuild": "0.0.0.8",
    "appNamespace": ".........",
    "wrapperRuntimeVersion": "12.6.0"
  }
}

**Comment by vitalyster** [comment]
As it was said before, most likely the issue is in server features: which XMPP server your clients are using? You need to reproduce the issue with the same server setup

**Comment by LumiProj** [comment]
Hi,
I have 4000k users I am using Ejabberd. There is no issue whatsoever Even my team tested and debug in iPhone 7 but no issue at all but some clients. particularly using iPhone 7 users are facing issue while connecting to xmpp server.
Sorry guys I am not able to provide the issue details.

**Comment by vitalyster** [comment]
Do they connect to your server and you are unable to debug your own server? Then what should we do?

**Comment by LumiProj** [comment]
Hi,
I am talking to the client side debugging. As far as server side debugging I have talked to Administrator to look into that and provide us the server side debug details.
I would share those details as soon as I have it.

Thanks
